### PR TITLE
fix(avatar-url-variables): variables not working with avatar url

### LINKED ISSUE #4081: [NV-2700] variables in avatar url field is not working
### What?

Our in-app editor has avatar option, There are three ways for adding avatar

1. User Avatar (novu will pick avatar from actor avatar URL)
2. System Avatar
3. Hard Coded Avatar URL

### Why? (Context)

If any user want to decouple avatar option from actor in first case, and don't want to use system avatar, only option left is to use hard coded avatar URL.

User does not have option to customize this via the payload variable 

image

### Definition of Done

1. User should be able to use variables avatar URL field and payload variable value should show correctly in in-app notiifcation

<sub>From [SyncLinear.com](https://synclinear.com) | [NV-2700](https://linear.app/novu/issue/NV-2700/variables-in-avatar-url-field-is-not-working)</sub>

### PR DESCRIPTION
**Pull request by rockingrohit9639**
fix #4081

### What change does this PR introduce?
Compiling variables in avatar url that user provides

### Why was this change needed?

Previously, the avatar URL was not working with variables

### Other information (Screenshots)

image

### CODE REVIEW

File: apps/worker/src/app/workflow/usecases/send-message/send-message-in-app.usecase.ts [hunk 1]
```diff
@@ -156,6 +156,17 @@ export class SendMessageInApp extends SendMessageBase {
...
+      if (actor && actor.data) {
+        actor.data = await this.compileInAppTemplate(
+          actor.data,
+          command.payload,
+          subscriber,
+          command,
+          organization,
+          tenant
+        );
+      }
```
Comment (p-fernandez) [inline apps/worker/src/app/workflow/usecases/send-message/send-message-in-app.usecase.ts:169]: @jainpawan21 this implementation will make that if user sets the avatar will always be shown but there could be cases where users don't want that to happen. Is this the expected behaviour?

Comment (rockingrohit9639) [reply apps/worker/src/app/workflow/usecases/send-message/send-message-in-app.usecase.ts:169]: I will update this

**Comment by davidsoderberg** [comment]
Would be great to have a api e2e test for this.

**Comment by davidsoderberg** [comment]
> > Would be great to have an api e2e test for this.
> 
> @davidsoderberg Could you please point Rohit to the file, where he should write the tests?

Yes of course, in `apps/api/src/app/events/e2e/trigger-event.e2e.ts` I think is the best place to do it in and there is also multiple example of how we do the testing :)

**Comment by rockingrohit9639** [comment]
> > > Would be great to have an api e2e test for this.
> > 
> > 
> > @davidsoderberg Could you please point Rohit to the file, where he should write the tests?
> 
> Yes of course, in `apps/api/src/app/events/e2e/trigger-event.e2e.ts` I think is the best place to do it in and there is also multiple example of how we do the testing :)

Thank you @davidsoderberg

**Comment by LetItRock** [comment]
hey @rockingrohit9639! 👋 thanks for your contribution 🤝 

Everything works well, but in the case when the URL contains characters like `&` or `=` they are "escaped" by the `CompileTemplate.execute` to the corresponding HTML entities, which results in this:
```
// before
https://gravatar.com/avatar/c1e91cf5bf83065c36134130164a0e3c?s=400&d=robohash&r=x

// after
https://gravatar.com/avatar/c1e91cf5bf83065c36134130164a0e3c?s&#x3D;400&amp;d&#x3D;robohash&amp;r&#x3D;x
```

Please take a look into this and please cover this functionality with the e2e tests. 
We really appreciate that! 🙌

**Comment by ainouzgali** [comment]
Hi @rockingrohit9639 , are you still working on it? :)

**Comment by hilmia** [comment]
Hi @rockingrohit9639 any updates on this? Would really appreciate this feature

**Comment by rockingrohit9639** [comment]
> Hi @rockingrohit9639 any updates on this? Would really appreciate this feature

Yes, working on it... will complete it in 1-2 days.

**Comment by jainpawan21** [comment]
@rockingrohit9639 following up on this PR 🙂

**Comment by hilmia** [comment]
Is there any way to use variables in the avatar? It would make things much easier.. Anyone working on this by any chance?

**Pull request closed without merging**